src/ui/TextFieldFund/TextFieldFund.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+
+import { Box, TextField, Text } from '../index';
+
+interface TextFieldFundProps {
+  fundsAvailable: number;
+  onChange(value: number): void;
+  value: number;
+  coinType?: string;
+}
+
+const parseFundToNumber = (value: string) => Number(value.replace(/,/g, ''));
+const parseFundToLocale = (value: number) => value.toLocaleString('en-US');
```

Comment:
This needs to be slightly different in order to let people type decimal values. Right now you can't type e.g. `12.50` because `12.` gets auto-converted to `12`